### DISCUSSION TITLE
Is there a way to get to the low level TinyUSB library C functions, bypassing the AdaFruit USB library?

### DISCUSSION THREAD

**Discussion by morio** [earlephilhower/arduino-pico#1790] [Q&A]
I've got a project compiling via CMake, but I'd like to move it to PlatformIO using this framework. I've got most of it compiling but it uses the TinyUSB library directly for USB host. 

Is there an easy way to access the C TinyUSB library that is a submodule of the pico-sdk which in turn is a submodule of this framework?

**Comment by earlephilhower** [comment]
You need to define NO_USB (double check the exact name, I'm traveling now without computer access) and then just include the appropriate headers in your code.  That's what the core's USB implementation does already, check RP2040USB.cpp.

You will lose the USB reset-to-upload and Serial, of course,  unless you manually reimplement then.
